Assignment_1/FileSearch.java

```diff
@@ -0,0 +1,77 @@
+package com.assignments.java;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+public class FileSearch {
+
+    /**
+     * The program searches through the home directory and looks for files that match a regular expression 
+     * and prints out the file name and its path.
+     * It takes the regular expression repeatedly as input.
+     *
+     * @param args
+     * @throws IOException
+     */
+    public static void main(String[] args) throws IOException {
+        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
+        File file = new File("/home");
+        Map<String, String> homeDirectoryMap = new HashMap<String, String>();
+        getAllFiles(file, homeDirectoryMap);
+
+        while (true) {
+        	AtomicInteger counter = new AtomicInteger();
```

Comment:
https://stackoverflow.com/questions/23201468/is-it-good-practice-to-use-atomicinteger-as-a-substitute-for-mutable-integer